### PR TITLE
HPCC-14732 Address Cleaner Plugin options with Configmgr

### PR DESCRIPTION
**Pull request by mckellyln**
This update allows for optional plugins from other packages to be configured properly.

@garonsky please review.
@Michael-Gardner please review.

Signed-off-by: Mark Kelly mark.kelly@lexisnexis.com

### CODE REVIEW

File: deployment/deployutils/deployutils.cpp [hunk 1]
```diff
@@ -827,6 +827,12 @@ class CGenerateJSFromXSD
               LoadComboBox("Software/DaliServerProcess", bAddBlank, m_pEnv, m_pEnv, strBuf);
               extraInfo = strBuf.str();
             }
+            else if (strcmp(type, "addrCleanerPluginType")==0)
```
Comment (mckellyln) [inline deployment/deployutils/deployutils.cpp]: This does reference the optional plugin by name but didn't see a way around that.

**Review by Michael-Gardner — APPROVED** [review]
Cmake looks good

File: deployment/deployutils/deployutils.cpp (hunk 1 above)
Comment (richardkchapman) [inline deployment/deployutils/deployutils.cpp]: Would be more generic if you tested for something like "type ends with PluginType" and then generated the xpath from by substituting PluginType for PluginProcess...

Comment (richardkchapman) [inline deployment/deployutils/deployutils.cpp]: Should also test that it's possible to support more than one plugin this way?

**Comment by garonsky** [comment]
looks good on my end.

File: deployment/deployutils/deployutils.cpp [hunk 2]
```diff
@@ -827,6 +827,15 @@ class CGenerateJSFromXSD
...
             }
+            else if (strstr(type, XML_TAG_PLUGINTYPE) != NULL)
+            {
+              StringBuffer pluginName(XML_TAG_SOFTWARE);
+              pluginName.append("/").append(type);
+              pluginName.replaceString(XML_TAG_PLUGINTYPE, XML_TAG_PLUGINPROCESS);
+              nCtrlType = 4;//LVC_COMBO;
+              LoadComboBox(pluginName.str(), bAddBlank, m_pEnv, m_pEnv, strBuf);
+              extraInfo = strBuf.str();
+            }
```
Comment (mckellyln) [inline deployment/deployutils/deployutils.cpp]: Generic approach to support any plugin following this convention.

File: deployment/deployutils/deployutils.cpp (hunk 1 above)
Comment (mckellyln) [inline deployment/deployutils/deployutils.cpp]: Updated code to be more generic.  Testing verified it works with multiple plugins.

**Comment by mckellyln** [comment]
@richardkchapman PR updated as per review comments.

**Pull request merged**